### PR TITLE
Code coverage

### LINKED ISSUE #30: Test coverage

### PR DESCRIPTION
**Pull request by kubaplas**
## Summary by CodeRabbit

- **Chores**
	- Improved CI workflow by adding code coverage reporting.
- **Refactor**
	- Modified `.gitignore` to exclude specific directories.
	- Added functionality to specify the target directory for build artifacts in `.cargo/config.toml` files.
- **New Features**
	- Added `llvm-tools-preview` component to the Rust toolchain.
	- Included steps to upload coverage to `codecov.io`.
	- Added `rustup component add llvm-tools-preview` and `cargo install grcov` in the `prepare-test-env` target.
	- Added a new `coverage` target for generating code coverage reports.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 .*
 *.env
 !/.gitignore
-/target/
 Cargo.lock
 .vscode/
 .idea/
 odra-casper/**/target/
-contract/
-examples/target
```
Comment (zie1ony) [inline .gitignore:12]: What's wrong with ignoring target of examples and modules?

File: examples/.cargo/config.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[build]
+target-dir = "../target"
```
Comment (zie1ony) [inline examples/.cargo/config.toml:2]: Why is this introduced?

**Review by zie1ony — APPROVED** [review]

File: examples/.cargo/config.toml (hunk 2 above)
Comment (kubaplas) [reply examples/.cargo/config.toml:2]: To merge code coverage from odra, examples and modules together
Also, it speeds up the compilation :3

File: .gitignore (hunk 1 above)
Comment (kubaplas) [reply .gitignore:12]: It is ignored already above

**Pull request merged**